components/nexus-core/src/main/java/org/sonatype/nexus/internal/blobstore/BlobStoreManagerImpl.java

```diff
@@ -51,29 +60,58 @@
 
   private final Path basedir;
 
-  private final Map<String,BlobStore> stores = Maps.newHashMap();
+  private final Map<String, BlobStore> stores = Maps.newHashMap();
+
+  private final ConfigurationStore store;
 
   @Inject
-  public BlobStoreManagerImpl(final ApplicationDirectories directories) {
+  public BlobStoreManagerImpl(final ApplicationDirectories directories, ConfigurationStore store) {
     checkNotNull(directories);
     this.basedir = directories.getWorkDirectory(BASEDIR).toPath();
+    this.store = store;
   }
 
-  // TODO: read configuration for blob-stores and start
+  @Override
+  protected void doStart() throws Exception { 
+    //TODO - appropriate place to insist on starting the store before this? new BlobStoreLifecycle impl, add to RepositoryLifecycle or...
+    store.start();
```

Comment:
@mcculls What is the appropriate way to ensure that this dependency is started before we get here?